### ISSUE TITLE
Better progress feedback for file operations

### ISSUE THREAD

**Issue by mherrmann** [fman-community/fman#46]
Currently, fman prints what files it is currently operating on in the status bar. It would be nicer to have (say) a progress bar. But should be unobtrusive/non-blocking. A user pointed out that time estimates are probably impossible to get right, so best to avoid them entirely.

**Comment by mherrmann** [comment]
Further requirements / nice to haves:
 * Any plugin should be able to display progress feedback via fman's API.
 * It would be nice if the user were also able to pause / abort the long running operations (if the plugin supports it).
 * The progress feedback should make it possible to display errors specific to that operation.

**Comment by mherrmann** [comment]
@joshas suggested maybe taking Filezilla as an inspiration for what the progress feedback could look like.

**Comment by webdesus** [comment]
Maybe better make core plugin? I think in a future fman have many plugins for copy use ftp protocol, smb, and etc. I think if we will be bet common interface for control copy operations, all future plugins created for  file operations, can use this common UI. What do you think about it?

**Comment by mherrmann** [comment]
@webdesus yes, that's the plan. There should be an API for displaying progress dialogs. Every plugin (including Core) can then use this API to show progress, handle cancellation etc.

**Comment by webdesus** [comment]
I get your point. you use Atom plugin Ideology))) Next question. Do you will make opportunity or it's task for community folks?

**Comment by mherrmann** [comment]
I will do it :)

**Comment by Greyvend** [comment]
Any updates on the progress feedback? I'd say it's the only minimally required file manager feature that fman currently lacks.

**Comment by mherrmann** [comment]
@Greyvend I'm currently working on a new file system backend that will make it possible to implement / fix #2, #30, #31, #32, #37 and many others. I agree with you that progress feedback is likely the most important "basic" feature that's still missing. I will implement it once I'm finished with the new file system backend.

**Comment by Greyvend** [comment]
@mherrmann any idea when it could be finished?

**Comment by mherrmann** [comment]
@Greyvend can't say precisely. Maybe Jan/Feb?

**Comment by mherrmann** [comment]
Implemented in [fman 1.1.2](https://fman.io/changelog#v1.1.2). Woohoo! :tada:

**Comment by joshas** [comment]
While file operations progress display is really needed addition, I was hoping for fman to implement it in more modern way, e.g. in the bottom part of window, to let user continue his work, while file operations are processing in the background. I agree, that in some cases that might make problems (working on files being deleted). How about adding "Move task to background" button, like in Total Commander?

**Comment by mherrmann** [comment]
@joshas feel free to open a new issue :-)
